bundles/org.eclipse.e4.ui.css.core/src/org/eclipse/e4/ui/css/core/util/impl/resources/HttpResourcesLocatorImpl.java

```diff
@@ -33,7 +33,7 @@ public String resolve(String uri) {
 
 	@Override
 	public InputStream getInputStream(String uri) throws Exception {
-		URL url = new java.net.URL((new File("./")).toURL(), uri);
+		URL url = new java.net.URL((new File("./")).toURI().toURL(), uri);
```

Comment:
This could have a few superfluous elements removed:
```suggestion
		URL url = new URL(new File("./").toURI().toURL(), uri);
```
Additionally I wonder if the context URL is even necessary at all?
The JavaDoc of the called `URL(URL context, String spec)` constructor says:
```
If the scheme component is defined in the given spec and does not match the scheme of the context,
then the new URL is created as an absolute URL based on the spec alone.
```
And since the passed string is definitively an `http`-URL and therefore has a scheme that is different from the `file`-scheme, I would say that the context is always discarded anyways.
So it could probably be just:
```suggestion
		URL url = new URL(uri);
```

Reply:
Makes sense, thanks for the suggestion.